### DISCUSSION TITLE
Question about expression context parsing

### DISCUSSION THREAD

**Discussion by ericsampson** [community/community#26337] [Actions]
<p>This might be the dumbest question ever…</p>
<p>What contexts in a workflow is it possible to use <span class="math">MYVAR and when it is required to use </span>{{ env.MYVAR }} ?</p>
<p>I can’t find any documentation on this.</p>

**Comment by ericsampson** [comment]
<p>also as a followup, is it possible to create a workflow level environment variable using bash to evaluate something dynamically?</p>

**Comment by airtower-luna** [comment, marked as answer]
<p><code>$MYVAR</code> is Bash (and generally Unix shell) syntax. <code>${{ env.MYVAR }}</code> is Github actions syntax. So you can use the former wherever you have a shell, which is <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstepsrun">in the <code>run</code> part of steps</a> (unless you set another interpreter, or run on Windows).</p>
<p>And for creating environment variables, see <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-commands-for-github-actions#setting-an-environment-variable">Setting an environment variable</a>. :wink: In short, you can do something like</p>
<pre><code class="lang-auto">echo "MYVAR=VALUE" &gt;&gt; $GITHUB_ENV
</code></pre>
<p>which writes the variable to a special file, and the Actions runner makes sure it’s available in <em>subsequent</em> steps. And you can of course build that definition dynamically.</p>

**Comment by ericsampson** [comment]
<p>Thanks! I wasn’t sure if <code>run</code> was the only step that supports shell syntax.</p>
<p>For the second question, I’m talking about trying to set a <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#env" rel="noopener nofollow ugc">workflow level env</a> variable using bash.</p>

**Comment by airtower-luna** [comment]
<p>There’s no equivalent at the workflow level. The syntax I mentioned above works at the job level, for all steps after the one setting the variable.</p>
<p>If you want to transfer data between jobs look at <a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs">outputs</a>, or possibly <a href="https://docs.github.com/en/free-pro-team@latest/actions/guides/storing-workflow-data-as-artifacts">artifacts</a>.</p>

**Comment by ericsampson** [comment]
<p>Thanks, too bad. I guess I’ll have to create a feature request for that!</p>
